frame/two-phase-election-provider/src/benchmarking.rs

```diff
@@ -0,0 +1,358 @@
+		T::Currency::make_free_balance_be(&receiver, initial_balance);
+		assert_ok!(T::Currency::reserve(&receiver, deposit));
+
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 10u32.into());
+	}: {
+		<TwoPhase<T>>::finalize_signed_phase_reject_solution(&receiver, deposit)
+	} verify {
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 0u32.into());
+	}
+
+	#[extra]
+	create_snapshot {
+		assert!(<TwoPhase<T>>::snapshot().is_none());
+	}: {
+		<TwoPhase::<T>>::create_snapshot()
+	} verify {
+		assert!(<TwoPhase<T>>::snapshot().is_some());
+	}
+
+	submit {
+		let c in 1 .. (T::MaxSignedSubmissions::get() - 1);
+
+		// the solution will be worse than all of them meaning the score need to be checked against all.
+		let solution = RawSolution { score: [(1000_0000u128 - 1).into(), 0, 0], ..Default::default() };
+
+		<CurrentPhase<T>>::put(Phase::Signed);
+		<Round<T>>::put(1);
+
+		for i in 0..c {
+			<SignedSubmissions<T>>::mutate(|queue| {
+				let solution = RawSolution { score: [(1000_0000 + i).into(), 0, 0], ..Default::default() };
+				let signed_submission = SignedSubmission { solution, ..Default::default() };
+				// note: this is quite tricky: we know that the queue will stay sorted here. The
+				// last will be best.
+				queue.push(signed_submission);
+			})
+		}
+
+		let caller = frame_benchmarking::whitelisted_caller();
+		T::Currency::make_free_balance_be(&caller,  T::Currency::minimum_balance() * 10u32.into());
+
+	}: _(RawOrigin::Signed(caller), solution, c)
+	verify {
+		assert!(<TwoPhase<T>>::signed_submissions().len() as u32 == c + 1);
+	}
+
+	submit_unsigned {
+		// number of votes in snapshot.
+		let v in 2000 .. 3000;
+		// number of targets in snapshot.
+		let t in 500 .. 800;
+		// number of assignments, i.e. compact.len(). This means the active nominators, thus must be
+		// a subset of `v` component.
+		let a in 500 .. 1500;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 200 .. 400;
+
+		let witness = WitnessData { voters: v, targets: t };
+		let raw_solution = solution_with_size::<T>(witness, a, d);
+
+		assert!(<TwoPhase<T>>::queued_solution().is_none());
+		<CurrentPhase<T>>::put(Phase::Unsigned((true, 1u32.into())));
+	}: _(RawOrigin::None, raw_solution, witness)
+	verify {
+		assert!(<TwoPhase<T>>::queued_solution().is_some());
+	}
+
+	// This is checking a valid solution. The worse case is indeed a valid solution.
+	feasibility_check {
+		// number of voters in snapshot.
+		let v in 200 .. 300;
+		// number of targets in snapshot.
+		let t in 80 .. 140;
+		// number of assignments, i.e. compact.voter_count(). This means the active nominators,
+		// thus must be a subset of `v` component.
+		let a in 80 .. 140;
+		// number of desired targets. Must be a subset of `t` component.
+		let d in 30 .. 60;
```

Comment:
why are those coef so different from submit_unsigned ?
Also the difference from submit_unsigned and feasiblity_check is huge and I can't really find why

Reply:
Probably I've forgotten to change them after doing a local test in which I wanted small numbers, will update.

Reply:
Changed it, but let's wait for the next benchmark run and then will resolve this if submit_unsigend and feasibility_check are of the same order.

Reply:
damn there is still a big difference,

The only difference I can see is that submit_unsigned is encoding `ReadySolution`. But this is unexpected to cost so much and also `v` shouldn't have any impact :-/

Reply:
the bot had errors and the numbers are not updated yet.

Reply:
Here's the new result. The coefficients seem similar but the dependency on `t` is strange to be nonexistent in one, although the coefficient is quite small I think.

Reply:
yes it looks fine now